internal/handler/msg_saslcontinue.go

```diff
@@ -44,11 +44,18 @@ func (h *Handler) MsgSASLContinue(ctx context.Context, msg *wire.OpMsg) (*wire.O
 
 	conv := conninfo.Get(ctx).Conv()
 
+	if conv == nil {
```

Comment:
Is it possible to add a test for calling saslcontinue directly? I'm not sure if that's possible

Reply:
It has to happen on an specific order or it'll fail for different reasons. I've tried a few strategies and failed so far to identify a way to do so that doesn't involve mimicking the wire protocol entirely (which looks complicated by what I've seen using a Wireshark sample of MongoDB data traffic) or finding a way to hijack the connection. Any ideas? /cc @b1ron

Reply:
I think capturing this scenario is hard in test is hard, so I'm happy as it is. But let's see what @b1ron says.

Reply:
Just curious about the solve. So this one line fixed the issue because `conv` is always nil when in proxy mode. But we only return responses from the proxy server thus we never return the error allowing us to complete the conversation.

Is my understanding correct?

cc. @henvic @chilagrow

Reply:
See the issue description and reproducible step in https://github.com/FerretDB/FerretDB/issues/4120.
This line avoids panic, that simple. It's because conversation was never stared in FerretDB. In that issue, what has happened is below.

Proxy: saslStart req-> saslStart ok -> saslContinue
FerretDB: saslStart req > saslStart not ok -> saslContinue panic

FerretDB failed because `username`/`password` with `SCRAM-SHA-265` does not exist